src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/regression_task_test.py

```diff
@@ -60,18 +60,47 @@ class TestcaseReproducesInRevisionTest(unittest.TestCase):
   def setUp(self):
     helpers.patch(self, [
         'clusterfuzz._internal.build_management.build_manager.setup_regular_build',
+        'clusterfuzz._internal.build_management.build_manager.check_app_path',
         'clusterfuzz._internal.bot.testcase_manager.test_for_crash_with_retries',
         'clusterfuzz._internal.bot.testcase_manager.check_for_bad_build',
     ])
 
   def test_error_on_failed_setup(self):
     """Ensure that we throw an exception if we fail to set up a build."""
     os.environ['APP_NAME'] = 'app_name'
+    regression_task_output = uworker_msg_pb2.RegressionTaskOutput()
+    self.mock.check_app_path.return_value = False
     # No need to implement a fake setup_regular_build. Since it's doing nothing,
     # we won't have the build directory properly set.
-    with self.assertRaises(errors.BuildSetupError):
-      regression_task._testcase_reproduces_in_revision(
-          None, '/tmp/blah', 'job_type', 1, should_log=False)
+    is_crash, error = regression_task._testcase_reproduces_in_revision(
+        None,
+        '/tmp/blah',
+        'job_type',
+        1,
+        regression_task_output,
+        should_log=False)
+    self.assertIsNone(is_crash)
+    self.assertIsNotNone(error)
+    self.assertEqual(error.error_type,
+                     uworker_msg_pb2.REGRESSION_BUILD_SETUP_ERROR)
+
+  def test_bad_build_error(self):
+    """Tests _testcase_reproduces_in_revision behaviour on bad builds."""
+    self.mock.check_app_path.return_value = True
+    build_data = uworker_msg_pb2.BuildData(
+        revision=1,
+        is_bad_build=True,
+        should_ignore_crash_result=False,
+        build_run_console_output='')
```

Comment:
WHy set this?

Reply:
`testcase_manager.check_for_bad_build` is supposed to return a build data. I'm mocking it here to make it return a bad build and then verify that the right info is passed back through the `regression_task_output`.

Reply:
I will remove `build_run_console_output=''` as it's unnecessary.